### DISCUSSION TITLE
Paskeys within embedded browser?

### DISCUSSION THREAD

**Discussion by lonoak** [passkeydeveloper/community#3] [General Q&A] [answered]
Hi,

   I'd like to ask for the problem I have seen in a couple of desktop applications using embedded ~~windows~~ browsers (in my case it's under macOS). Passkeys are being used as a second factor here (in a SAML identity provider), and fails in apps that support an external authentication service, which sometimes let you use directly (for example in Zoom as an Outlook plugin), or indirectly, (usually via Google or Microsoft) your own identity provider for authenticating. 

<img width="996" alt="imagen" src="https://github.com/passkeydeveloper/community/assets/2891452/c057b0a4-bbea-46ab-af54-5e39a587e9e9">

Has anyone seen this issue or knows if there's a way to use passkeys within embedded browsers in macOS?

Thanks in advance!

Jose.

**Comment by lonoak** [comment]
Answering to myself, I see embedded webviews are marked unsupported in the [passkeys.dev](https://passkeys.dev/device-support/#device-support-table) website, so I guess it's not yet in the priority list of OS vendors...

**Reply by timcappalli** [reply]
@lonoak So it is a bit complicated with services who use federated providers, which is why they are generically listed as not supported. EWV's run in a first party context to the calling app, so only passkeys that are bound to the web origin of the calling app can be used. This is by design as EWVs can modify all parts of the web content. You wouldn't want app A mucking with or observing passkey sign ins from origin B.

System WebViews (or popping a default browser) should be used in cases where you have multiple identity providers for sign in, as those providers may be using passkeys for sign in.

**Reply by lonoak** [reply]
Thanks @timcappalli.

I later realized these apps should better implement RFC 8252 BP for delegating authentication to the browser. I guess this would solve any security problems, as well not break passkeys.

**Comment by timcappalli** [comment, marked as answer]
passkeys.dev/device-support as well as the platform reference pages now include details about EWV and SWV support for passkeys.

**Reply by sc3110308-tech** [reply]
sc3110307@gmail.com

**Reply by sc3110308-tech** [reply]
> sc3110307@gmail.com 
> 

Password

**Reply by phattrantan166-svg** [reply]
> Trang passkeys.dev/device-support cũng như các trang tham khảo nền tảng hiện đã bao gồm thông tin chi tiết về hỗ trợ EWV và SWV cho passkey.

**Comment by phattrantan166-svg** [comment]
> CHÀO,
> 
> Tôi muốn hỏi về vấn đề mà tôi đã gặp phải trong một vài ứng dụng máy tính để bàn sử dụng mã nhúng.~cửa sổ~Trình duyệt (trong trường hợp của tôi là macOS). Passkey được sử dụng như một yếu tố xác thực thứ hai (trong nhà cung cấp danh tính SAML), và không hoạt động trong các ứng dụng hỗ trợ dịch vụ xác thực bên ngoài, đôi khi cho phép bạn sử dụng trực tiếp (ví dụ: trong Zoom như một plugin của Outlook), hoặc gián tiếp (thường thông qua Google hoặc Microsoft), nhà cung cấp danh tính của riêng bạn để xác thực.
> 
> <img alt="hình ảnh" width="996" src="https://private-user-images.githubusercontent.com/2891452/308615647-c057b0a4-bbea-46ab-af54-5e39a587e9e9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODQ0MzU4NDQsIm5iZiI6MTc4NDQzNTU0NCwicGF0aCI6Ii8yODkxNDUyLzMwODYxNTY0Ny1jMDU3YjBhNC1iYmVhLTQ2YWItYWY1NC01ZTM5YTU4N2U5ZTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDcxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjA3MTlUMDQzMjI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjZlZmQ2N2YyOWI3YjEzNzRkY2ZiYTU4NDE4NTQ4ZDg3ZmJiZTg2OTkzOTc4N2M1N2ZmNWEwOTZhOWJiYjkyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmcmVzcG9uc2UtY29udGVudC10eXBlPWltYWdlJTJGcG5nIn0.HgOou2i_LW969vYXU1SsRPwgLoEZTeCxDNSObw-PMxA">
> Có ai gặp phải vấn đề này hoặc biết cách sử dụng mật khẩu trong trình duyệt nhúng trên macOS không?
> 
> Cảm ơn trước!
> 
> Jose.

https://github.com/orgs/passkeydeveloper/discussions/3#discussion-6291299
[Uploading 1735866528953b70447a83b524125b7456589c2cdc7f3.pdf…]()
